File: output/ChatGPT/avatar/Python/Java/codeforces_30_A.java
public class codeforces_30_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int A = scanner.nextInt();
        int B = scanner.nextInt();
        int n = scanner.nextInt();
        int result = task4(A, B, n);
        System.out.println(result);
    }
    
    public static int task4(int A, int B, int n) {
        for (int X = -1000; X <= 1000; X++) {
            if (A * Math.pow(X, n) == B) {
                return X;
            }
        }
        return -1; // No solution
    }
}

File: output/ChatGPT/avatar/Python/C++/codeforces_30_A.cpp
#include <iostream>
#include <cmath>

using namespace std;

int task4(int A, int B, int n) {
    for (int X = -1000; X <= 1000; X++) {
        if (A * pow(X, n) == B) {
            return X;
        }
    }
    return -1; // No solution
}

int main() {
    int A, B, n;
    cin >> A >> B >> n;
    cout << task4(A, B, n) << endl;

    return 0;
}